php_uv.c

```diff
@@ -341,17 +341,18 @@ static php_socket_t php_uv_zval_to_valid_poll_fd(zval *ptr)
 	/* TODO: is this correct on windows platform? */
 	if (Z_TYPE_P(ptr) == IS_RESOURCE) {
 		if (ZEND_FETCH_RESOURCE_NO_RETURN(stream, php_stream *, ptr, -1, NULL, php_file_le_stream())) {
-			/* make sure only valid resource streams are passed - plainfiles and php streams are invalid */
+			/* Some streams (specifically STDIO and encrypted streams) can be cast to FDs */
+			if (php_stream_cast(stream, PHP_STREAM_AS_FD_FOR_SELECT | PHP_STREAM_CAST_INTERNAL, (void*)&fd, 1) == SUCCESS && fd >= 0) {
+				return fd;
+			}
 			if (stream->wrapper) {
```

Comment:
why is this `if` required at all? Only to make the `E_ERROR` appear to ease debugging?